chapter02/worksheets/solution2.1.7.2_AR.scala

```diff
@@ -0,0 +1,17 @@
+/** author: Andreas Röhler */
+
+/** 
+  Exercise 2.1.7.2
+Find al triples i, j, k where 0 ≤ i,j,k ≤ 9 and i + 4 ∗ j + 9 * k > i ∗ j * k.
+Hint: use flatMap and filter.
+  */
+
+val a = (0 to 9).flatMap(x => (0 to 9).flatMap(y => (0 to 9).map { z => (x, y, z) } ))
+val b = a.map{ case (i, j, k) => (i +  4 * k + 9 * j) }
+val c = a.map{ case (i, j, k) => (i * j * k) }
+val d = b.zip(c)
+val result = d.filter { case (x, y) => x > y }
```

Comment:
The result should be a list of triples. You are producing a list of tuples containing the intermediate results.